mantidproject/mantid
# Reset SANS user file on loading failure

**Description of work.**
If loading a user file fails for any reason, the user file field is reset to an empty string to make it clear that the would-be user file was not loaded

**Report to:** nobody

**To test:**
[sans2d_user_file.txt](https://github.com/mantidproject/mantid/files/2813081/sans2d_user_file.txt)
1. Interfaces -> SANS -> SANS v2
2. Click load user file and select the attached file. This should succeed without warning
3. Close the GUI and reopen it. This user file should already be pre-loaded
4. Click load user file again and select any file which isn't a legitimate sans user file, e.g. a csv file. A popup should explain that loading of the user file has failed, and the user file text field should be empty
5. Close and re-open the GUI again. The user file text field should be empty and you should see no errors on the MantidPlot logger regarding user file loading (it should not have tried to load an empty user file)

Fixes #24446 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
